### PR TITLE
Encryption spec update

### PR DESCRIPTION
**Pull request by mattheworiordan**
I've gone ahead and rolled out the changes to the spec in Ruby as well which fixes https://github.com/ably/ably-ruby/issues/80

I discovered a few bugs along the way so fixed everything that needed doing before implementing the Crypto changes.

cc @SimonWoolf

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -157,6 +157,19 @@ channel.presence.history do |presence_page|
...
+
+```ruby
+secret_key = Ably::Util::Crypto.generate_random_key
+channel = client.channels.get('test', cipher: { key: secret_key })
+channel.subscribe do |message|
+  message.data #=> "sensitive data"
+end
+channel.publish nil, "sensitive data" # data will be encrypted before publish
+```
+
```
Comment (SimonWoolf) [inline README.md]: Suggestion: could make it clear that the name isn't encrypted? eg
```ruby
secret_key = Ably::Util::Crypto.generate_random_key
channel = client.channels.get('test', cipher: { key: secret_key })
channel.subscribe do |message|
  message.data #=> "sensitive data (encrypted before being published)"
end
channel.publish "name (not encrypted)", "sensitive data (encrypted before being published)"
```

**Comment by SimonWoolf** [comment]
LGTM :)

**Comment by mattheworiordan** [comment]
> Is a combined option something that we want to support in general? (I haven't included it in ably-js, and it isn't in the spec)

I don't think it needs to be part of the spec, I opted to include it in the Ruby lib

File: README.md (hunk 1 above)
Comment (mattheworiordan) [inline README.md]: Sure, will do

File: README.md [hunk 2]
```diff
@@ -157,6 +157,19 @@ channel.presence.history do |presence_page|
 end
 ```
 
+### Symmetrical end-to-end encrypted payloads on a channel
```
Comment (paddybyers) [inline README.md]: I think it should be symmetric encryption, not symmetrical.

Comment (mattheworiordan) [inline README.md]: Yup thanks

**Pull request merged**